added san diego components

Steve-Dusty opened the pull request.

samdev-7 reviewed (changes requested):
Please put this in the same pr as the main san-diego.js change.

Steve-Dusty commented:
> Please put this in the same pr as the main san-diego.js change.

I removed the previous pull request, and added the san-diego.js in this pull request.

samdev-7 commented:
Closing this due to inactivity. Feel free to re-open if you still want these changes to be added.
